src/pool/pool_disjoint.c

```diff
@@ -819,6 +820,11 @@ umf_result_t disjoint_pool_free(void *pool, void *ptr) {
             return ret;
         }
 
+        if (pool != umfPoolGetPoolPriv(allocInfo.pool)) {
```

Comment:
Why do we need this check? It breaks encapsulation because now pool implementation calls `umfPoolGetPoolPriv`.

Reply:
This check prevents from freeing a pointer from a different pool.

Reply:
@vinser52 It is tested in https://github.com/oneapi-src/unified-memory-framework/pull/1143
See https://github.com/oneapi-src/unified-memory-framework/pull/1143#discussion_r1981344299

Reply:
@vinser52 It is needed by https://github.com/oneapi-src/unified-memory-framework/pull/1143
See https://github.com/oneapi-src/unified-memory-framework/pull/1143#discussion_r1981344299

Reply:
This PR was replaced with https://github.com/oneapi-src/unified-memory-framework/pull/1161 - please review this new PR instead.